scenario_player/main.py

```diff
@@ -242,29 +268,25 @@ def pack_logs(post_to_rocket, raiden_dir, pack_n_latest, scenario_names, target_
         post_to_rocket_chat(archive_fpath)
 
 
-def pack_n_latest_logs_for_scenario_in_dir(scenario_name, scenario_log_dir: Path, n) -> set:
-    # Add all scenario logs requested. Drop any iterations older than pack_n_latest,
-    # or add all if that variable is 0.
-    scenario_logs = [path for path in scenario_log_dir.iterdir() if (path.is_file() and str(path).startswith(scenario_name))]
-    scenario_logs = sorted(scenario_logs, key=lambda x: x.stat().st_mtime, reverse=True)
-
-    # specifying `pack_n_latest=0` will add all scenarios.
-    max_range = n or len(scenario_logs)
+def pack_n_latest_logs_for_scenario_in_dir(scenario_name, scenario_log_dir: Path, n) -> list:
+    """Add the `n` latest log files for `scenario_name` in `scenario_dir` to a :cls:`set` and return it."""
+    scenario_logs = [
+        path for path in scenario_log_dir.iterdir()
+        if (path.is_file() and str(path).startswith(scenario_name))
+    ]
+    history = sorted(scenario_logs, key=lambda x: x.stat().st_mtime, reverse=True)
 
-    files = set()
+    # Can't pack more than the number of available logs.
+    num_of_packable_iterations = range(n or len(scenario_logs))
 
-    for i in range(max_range):
-        try:
-            files.add(scenario_logs[i])
-        except IndexError:
-            # We ran out of scenario logs to add before reaching the requested number of n latest logs.
-            print(
-                f'Only packed {i} logs of requested latest {n} '
-                f'- no more logs found for {scenario_name}!',
-            )
-            break
+    if num_of_packable_iterations < n:
+        # We ran out of scenario logs to add before reaching the requested number of n latest logs.
+        print(
+            f'Only packing {num_of_packable_iterations} logs of requested latest {n} '
+            f'- no more logs found for {scenario_name}!',
+        )
 
-    return files
+    return history[:num_of_packable_iterations]
```

Comment:
Depending how (or if at all) log rotation / purging is done, it could with time become expensive to build a list of all logs and then throwing most of it away. 

Can be improved alter of course.